lib/Net/SAML2/Binding/POST.pm

```diff
@@ -76,4 +81,47 @@ sub handle_response {
     return $xml;
 }
 
+=head2 prepare_request( $request )
+
+Sign and encode the SAMLRequest.
+
+=cut
+
+sub prepare_request {
+    my ($self, $request) = @_;
+
+    my $signer = XML::Sig->new({
+                        key => $self->{key},
```

Comment:
Please try to use: $self->key and $self->cert instead of $self->{key}, $self->{cert}.

If we change the way the attributes (or methods) key/cert are set we need to update this bit of code.

Reply:
will do